src/poliastro/plotting/porkchop.py

```diff
@@ -23,231 +20,241 @@
-    >>> dv_launch, dev_dpt, c3dpt, c3arr, tof = porkchop(Earth, Mars, launch_span, arrival_span)
-    """
-
-    dv_launch, dv_arrival, c3_launch, c3_arrival, tof = targetting_vec(
+    def __init__(
+        self,
         departure_body,
         target_body,
-        launch_span[np.newaxis, :],
-        arrival_span[:, np.newaxis],
-    )
-
-    # Start drawing porkchop
-
-    if ax is None:
-        fig, ax = plt.subplots(figsize=(15, 15))
-    else:
-        fig = ax.figure
-
-    c3_levels = np.linspace(0, max_c3.to(u.km ** 2 / u.s ** 2).value, 30)
-
-    c = ax.contourf(
-        [D.to_datetime() for D in launch_span],
-        [A.to_datetime() for A in arrival_span],
-        c3_launch,
-        c3_levels,
-    )
-
-    line = ax.contour(
-        [D.to_datetime() for D in launch_span],
-        [A.to_datetime() for A in arrival_span],
-        c3_launch,
-        c3_levels,
-        colors="black",
-        linestyles="solid",
+        launch_span,
+        arrival_span,
+        ax=None,
+        tfl=True,
+        vhp=True,
+        max_c3=45.0 * u.km ** 2 / u.s ** 2,
+        max_vhp=5 * u.km / u.s,
+    ):
+        """
+        Parameters
+        ----------
+        departure_body: poliastro.bodies.Body
+            Body from which departure is done
+        target_body: poliastro.bodies.Body
+            Body for targetting
+        launch_span: astropy.time.Time
+            Time span for launch
+        arrival_span: astropy.time.Time
+            Time span for arrival
+        ax: matplotlib.axes.Axes:
+            For custom figures
+        tfl: boolean
+            For plotting time flight contour lines
+        vhp: boolean
+            For plotting arrival velocity contour lines
+        max_c3: float
+            Sets the maximum C3 value for porkchop
+        max_vhp: float
+            Sets the maximum arrival velocity for porkchop
+        """
+        self.departure_body = departure_body
+        self.target_body = target_body
+        self.launch_span = launch_span
+        self.arrival_span = arrival_span
+        self.ax = ax
+        self.tfl = tfl
+        self.vhp = vhp
+        self.max_c3 = max_c3
+        self.max_vhp = max_vhp
+
+    def _get_state(self, body, time):
+        """ Computes the position of a body for a given time. """
+
+        solar_system_bodies = [
+            Sun,
+            Mercury,
+            Venus,
+            Earth,
+            Moon,
+            Mars,
+            Jupiter,
+            Saturn,
+            Uranus,
+            Neptune,
+            Pluto,
+        ]
+
+        # We check if body belongs to poliastro.bodies
+        if body in solar_system_bodies:
+            rr, vv = coord.get_body_barycentric_posvel(body.name, time)
+        else:
+            rr, vv = body.propagate(time).rv()
+            rr = coord.CartesianRepresentation(rr)
+            vv = coord.CartesianRepresentation(vv)
+
+        return rr.xyz, vv.xyz
+
+    def _targetting(self):
```

Comment:
I think we should keep this function outside of the class, it's useful on its own regardless of Porkchop plots.

Reply:
Yes this will fix it

Reply:
I tried multiple permutations for resolving the issue but I think what this will do is that it will fix the issue of the parameters passing around !!

Reply:
Done, Works as intended :) Thank you very much @astrojuanlu for this review :) 
Thank you @jorgepiloto for giving me this chance :)